template-coq/src/constr_denoter.ml

```diff
@@ -210,32 +210,35 @@ struct
     else
       not_supported_verb trm "unquote_level"
 
-  let unquote_univ_expr evm trm (* of type UnivExpr.t *) : Evd.evar_map * Univ.Universe.t =
+  let unquote_univ_expr evm trm (* of type UnivExpr.t *) : Evd.evar_map * Sorts.t =
     let (h,args) = app_full trm [] in
     if constr_equall h c_pair then
       let l, b = unquote_pair trm in
       let evm, l' = unquote_level evm l in
-      let u = Univ.Universe.make l' in
-      evm, if unquote_nat b > 0 then Univ.Universe.super u else u
+      let u = Sorts.sort_of_univ @@ Univ.Universe.make l' in
+      evm, if unquote_nat b > 0 then Sorts.super u else u
     else
       not_supported_verb trm "unquote_univ_expr"
 
+  let sort_sup s1 s2 =
+    Sorts.sort_of_univ (Univ.Universe.sup (Sorts.univ_of_sort s1) (Sorts.univ_of_sort s1))
```

Comment:
This is somehow buggy, I put some printf debugging
~~~ocaml
  let sort_sup s1 s2 =
    let s = Sorts.sort_of_univ (Univ.Universe.sup (Sorts.univ_of_sort s1) (Sorts.univ_of_sort s1)) in
    Feedback.msg_info Pp.(str "s1=" ++ Sorts.debug_print s1 ++ str ", s2=" ++ Sorts.debug_print s2 ++ str ", s=" ++ Sorts.debug_print s);
    s
~~~
and it said
~~~
s1=Set, s2=Type(Var(0)), s=Set
~~~
(should be `s=max(Set,Var 0)`

Reply:
```suggestion
    Sorts.sort_of_univ (Univ.Universe.sup (Sorts.univ_of_sort s1) (Sorts.univ_of_sort s2))
```
x_x

Reply:
Ach!